lib/std/heap.zig

```diff
@@ -8,6 +8,273 @@ const c = std.c;
+    },
+    .netbsd => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // Alpha
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => missing_min_page_size,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/mips/include/vmparam.h#L73
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/powerpc/include/vmparam.h#L41
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => 4 << 10,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/sparc/include/vmparam.h#L69
+        .sparc => 4 << 10,
+        .sparc64 => missing_min_page_size,
+        .riscv32, .riscv64 => missing_min_page_size,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/m68k/include/vmparam.h#L70
+        .m68k => 4 << 10,
+        else => missing_min_page_size,
+    },
+    .dragonfly => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        else => missing_min_page_size,
+    },
+    .openbsd => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // Alpha
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => missing_min_page_size,
+        .mips, .mipsel, .mips64, .mips64el, .loongarch32, .loongarch64 => missing_min_page_size,
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => missing_min_page_size,
+        .riscv64 => missing_min_page_size,
+        .sparc64 => missing_min_page_size,
+        else => missing_min_page_size,
+    },
+    .linux => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 4 << 10,
+        // Explicitly only 4kb.
+        // https://refspecs.linuxbase.org/ELF/zSeries/lzsabi0_zSeries.html#AEN798
+        .s390x => 4 << 10,
+        // Source: Wikipedia "MIPS" and Linux mips/Kconfig.
+        .mips, .mipsel, .mips64, .mips64el, .loongarch32, .loongarch64 => 4 << 10,
+        // Source: csky/Kconfig only selects HAVE_PAGE_SIZE_4KB.
+        .csky => 4 << 10,
+        // Hexagon's Kconfig/page.h in Linux. Non-huge pages go past 256KB.
+        .hexagon => 4 << 10,
+        // Source: Zig's own libc page.h for arc.
+        .arc => 4 << 10,
+        // Source: Wikipedia "Page (computer memory)"
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => 4 << 10,
+        .riscv32, .riscv64 => 4 << 10,
+        .sparc => 4 << 10,
+        .sparc64 => 8 << 10,
+        else => missing_min_page_size,
+    },
+    else => missing_min_page_size,
+};
+
+/// This value defines the largest page size for this architecture/OS combination that the standard library allows. The standard library asserts that the `pageSize()` does not exceed `max_page_size`. To allow larger page sizes, override `max_page_size` as well as `-z max-page-size`.
```

Comment:
> To allow larger page sizes, override `max_page_size`

This should probably be reworded to something like: To allow larger page sizes, add or modify an appropriate prong in the `max_page_size` definition.

The previous wording makes it sound like the user can actually influence the resulting value externally.

> as well as `-z max-page-size`.

Does this actually affect `mmap` and friends? That would be surprising to me. I don't get the impression that this linker option has anything to do with that. cc @kubkon

Reply:
`-z max-page-size` is a very similar list to ours found in the LLVM linker. It's not used by `mmap` but it is used to align ELF sections, so it induces the same sort of crash by unalignment. Unlike `max_page_size` it simply segfaults when exceeded, so it felt appropriate to note.